### PR TITLE
Fix navigation to Concierge chat after sign-up/sign-in

### PR DESCRIPTION
**Pull request by marcochavezf**
### Details

This PR fixes the navigation to Concierge right a user signs up or signs in after opening the link https://new.expensify.com/concierge

### Fixed Issues

$ https://github.com/Expensify/App/issues/15785
$ https://github.com/Expensify/App/issues/16080

### Tests

1. Log out if you're already logged in
1. Open https://new.expensify.com/concierge
1. Sign-up with a new account
1. Verify it navigates to the Concierge chat
1. Sign-out, close the app / refresh the browser
1. Open again https://new.expensify.com/concierge
1. Sign-in with an existing account
1. Verify again it navigates to the Concierge chat
1. Log out, don't close the app
1. Open again https://new.expensify.com/concierge
1. Sign-in with an existing account
1. Verify again it navigates to the Concierge chat

To navigate to https://new.expensify.com/concierge on dev:

- On web just paste the URL in the browser (i.e. http://localhost:8080/concierge) and click on open this link in your browser
- On desktop also paste the URL in the browser but click on Open Electron from the alert dialog
- On Android open the deep link from Terminal: npx uri-scheme open new-expensify://concierge--android
- On iOS also open the deep link from Terminal: npx uri-scheme open new-expensify://concierge--ios

To navigate to https://new.expensify.com/concierge on staging/production:

- On web just paste the URL in the browser (i.e. https://new.expensify.com/concierge) and click on open this link in your browser
- On desktop also paste the URL in the browser but click on Open Electron from the alert dialog
- On native open the report chat from an external link (for example send https://staging.new.expensify.com/concierge to your email and open the link from the phone).

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as the test steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/6829422/224151386-bc5d24d5-08fd-446f-b3c8-3e46d5a5911a.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/6829422/224153843-fd414ad6-d96b-49b2-bf84-c73ae939145c.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/6829422/224152246-f958f76c-3ecb-42cf-94b9-17e8d2025e41.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/6829422/224156802-ff711e40-ef21-494e-823b-412f77772753.mov

After signing out without closing the application:

https://user-images.githubusercontent.com/6829422/226499968-133cd8c0-56f0-4664-ba37-c7a030ed074a.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/6829422/224156367-29f2e70b-e911-439e-8893-21b2ecf9ec8a.mov

After signing out without closing the application:

https://user-images.githubusercontent.com/6829422/226499827-4ed2a73c-8408-4edc-8a71-d9f6ff5fe6a7.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/6829422/224155991-1ce40390-b57b-4407-b889-5909bb1531ea.mov

After signing out without closing the application:

https://user-images.githubusercontent.com/6829422/226499632-969ddc71-9b76-42d1-89f7-1a7d6680fe78.mov

</details>

### CODE REVIEW

**Comment by danieldoglas** [comment]
@thesahindia can you please test and do the checklist here?

**Comment by marcochavezf** [comment]
Closing it in favor of this [proposal](https://github.com/Expensify/App/issues/15785#issuecomment-1464856926).

**Comment by marcochavezf** [comment]
Hi @thesahindia, I didn't see your previous comments and didn't notice you already reviewed the PR, sorry about that 🙇🏽 I think I didn't refresh the PR when I posted this [comment](https://github.com/Expensify/App/pull/15788#issuecomment-1469241447).

I'm re-opening this PR because there is a [regression](https://github.com/Expensify/App/issues/16080) with the other proposal. I will update the steps and add some videos covering the regression (only reproducible on native). Could you test it again? Thanks

**Comment by marcochavezf** [comment]
> Tests went well with an existing account but not with a new account. I didn't get navigated to concierge. The fab menu opens for the new account.
> 
> cc: @marcochavezf

Btw, I tested again and it's working with new accounts when I open the app from the deep link.

**Review by mountiny — APPROVED** [review]
Looks good to me, in sake of moving the deploy forward, I will merge this to get it to staging. Tests seem good to me in the PR description

**Comment by isagoico** [comment]
Retest for this CP was a pass in the following platforms:

- Web/mWeb
- Android
- Desktop

Note: For iOS, the deep linking is broken so we were blocked from testing on that platform - issue here https://github.com/Expensify/App/issues/15158

Checking it off the list

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -1367,6 +1367,24 @@ function toggleEmojiReaction(reportID, reportAction, emoji, paramSkinTone = pref
...
+    InteractionManager.runAfterInteractions(() => {
+        Navigation.isReportScreenReady().then(() => {
+            const route = ReportUtils.getRouteFromLink(url);
+            const reportID = ReportUtils.getReportIDFromLink(url);
+            if (reportID) {
+                Navigation.navigate(ROUTES.getReportRoute(reportID));
+            }
+            if (route === ROUTES.CONCIERGE) {
+                navigateToConciergeChat();
+            }
```
Comment (mollfpr) [inline src/libs/actions/Report.js:1383]: ✋ Coming from https://github.com/Expensify/App/issues/23745

We are not handling the navigation from other URLs except the report or concierge here.

**Pull request merged**